impl/src/main/java/org/jboss/cdi/tck/tests/extensions/configurators/bean/AfterBeanDiscoveryObserver.java

```diff
@@ -94,84 +91,49 @@ public void observeUndead(@Observes AfterBeanDiscovery abd, BeanManager bm) {
         // make passivation capable
         zombie.id("zombie");
 
+        // create Ghost bean
         BeanConfigurator<Ghost> ghost = abd.addBean();
 
+        // creation using read from bean attributes
+        ghost.read(bm.createBeanAttributes(bm.createAnnotatedType(Ghost.class)));
         ghost.beanClass(Ghost.class);
         ghost.addQualifier(Undead.UndeadLiteral.INSTANCE);
         ghost.addStereotype(Monster.class);
-
-        injectionPointBuilder.configure().read(ghostFieldInDungeon)
-            .addQualifier(Undead.UndeadLiteral.INSTANCE)
-            .type(Ghost.class);
-
-        InjectionPoint dungeonIP = injectionPointBuilder.build();
-
-        injectionPointBuilder.configure().read(ghostFieldInTower)
-            .addQualifier(Undead.UndeadLiteral.INSTANCE)
-            .type(Ghost.class);
-
-        InjectionPoint towerIP = injectionPointBuilder.build();
-
-        // add multiple IPs
-        ghost.addInjectionPoints(dungeonIP, towerIP);
+        ghost.scope(RequestScoped.class);
+
+        // test replacement of IPs
+        InjectionPoint ghostWeaponIP = null;
```

Comment:
This variable is not necessary here (nothing will change when I remove it). I would make this a bit more readable and split this large `observeUndead` method into some separated methods (e.g configureSkeleton).